### PR TITLE
Standard token and nft richer logs

### PR DESCRIPTION
**Pull request by dangershony**

### CODE REVIEW

File: CRS/Cirrus.Node/Controllers/IndexerContractsController.cs [hunk 1]
```diff
@@ -70,7 +74,7 @@ public IActionResult GetContractInfo([FromQuery] string txHash)
                 typeName = this.stateRoot.GetContractType(address);
             }
 
-            List<LogResponse> logResponses = null;
+            List<LogResponse> logResponses = new List<LogResponse>();
```
Comment (sondreb) [inline CRS/Cirrus.Node/Controllers/IndexerContractsController.cs:78]: This is good and ensures empty array is returned as oppose to null value. Methods and APIs should return empty arrays and not null when there is no results.

File: CRS/Cirrus.Node/Controllers/IndexerContractsController.cs [hunk 2]
```diff
@@ -79,11 +83,76 @@ public IActionResult GetContractInfo([FromQuery] string txHash)
...
-            return this.Json(new ContractReceiptResponse(receipt, logResponses ?? new List<LogResponse>(), this.network)
+            EnrischLogs(receipt, typeName, logResponses);
+
+            return this.Json(new ContractReceiptResponse(receipt, logResponses, this.network)
             {
                 ContractCodeType = typeName,
             });
         }
 
+        private void EnrischLogs(Receipt receipt, string typeName, List<LogResponse> logResponses)
```
Comment (sondreb) [inline CRS/Cirrus.Node/Controllers/IndexerContractsController.cs]: EnrichLogs, typo in the method name

File: CRS/Cirrus.Node/Controllers/IndexerContractsController.cs [hunk 3]
```diff
@@ -79,11 +83,76 @@ public IActionResult GetContractInfo([FromQuery] string txHash)
...
+        {
+            if (typeName == "StandardToken")
```
Comment (sondreb) [inline CRS/Cirrus.Node/Controllers/IndexerContractsController.cs]: Please make an static variable or enum for these, so you can do "ContractTypeName.StandardToken", etc.

File: CRS/Cirrus.Node/Controllers/IndexerContractsController.cs [hunk 4]
```diff
@@ -79,11 +83,76 @@ public IActionResult GetContractInfo([FromQuery] string txHash)
...
+                if (receipt.Success)
+                {
+                    if (receipt.MethodName == null && receipt.NewContractAddress != null)
+                    {
+                        // this is the constructor we want to fetch the name, symbol, decimals and total supply.
+                        IStateRepositoryRoot stateAtHeight = this.stateRoot.GetSnapshotTo(receipt.PostState.ToBytes());
+
+                        uint160 addressNumeric = receipt.NewContractAddress;
+
+                        byte[] tokenName = stateAtHeight.GetStorageValue(addressNumeric, Encoding.UTF8.GetBytes("Name"));
```
Comment (sondreb) [inline CRS/Cirrus.Node/Controllers/IndexerContractsController.cs]: Encoding.UTF8 could be replaced with const encoding = Encoding.UTF8 and then referenced all places below. I'm not sure if the compiler actually might optimize this automatically.

**Review by sondreb — APPROVED** [review]
Approved with suggestions

File: CRS/Cirrus.Node/NonFungibleTokenLogEnrichment.cs [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+        const string typeName = "NonFungibleToken";
+
+        public bool IsSmartContractSupported(string typeName)
+        {
+            return typeName == NonFungibleTokenLogEnrichment.typeName;
+        }
+
+        public void EnrichLogs(Receipt receipt, List<LogResponse> logResponses)
+        {
```
Comment (dangershony) [inline CRS/Cirrus.Node/NonFungibleTokenLogEnrichment.cs]: This only enriches the constructor trx, you might want to mae logs on other methods will they go here or will they get a new class?

**Pull request merged**